pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -254,10 +259,10 @@ func (r *MachineSyncReconciler) Reconcile(ctx context.Context, req reconcile.Req
 		return ctrl.Result{}, nil
 	case authoritativeAPI == "":
 		logger.Info("Machine status.authoritativeAPI is empty, will check again later", "AuthoritativeAPI", mapiMachine.Status.AuthoritativeAPI)
-		return ctrl.Result{}, nil
+		return ctrl.Result{RequeueAfter: requeueUnknownAuthoritativeAPI}, nil
```

Comment:
Have I missed something here? Why is this required? This controller watches Machine, so it will be reconciled immediately if `status.authoritativeAPI` is set anyway.